File: Codility/Codility-js/2-arrays/OddOccurrencesInArray.js
function oddOccurrance(A){
    A = A.sort();
    var count = 0;
    for (var i = 0; i < A.length; i++){
       count++;
       if(A[i] !== A[i + 1]){
           if(count % 2 === 1) return A[i];
           count = 0;
       }
    }
}

console.log(oddOccurrance([20, 2, 3, 6, 6, 3, 2, 7 ,2, 2, 7, 20, 12, 12, 12]));
console.log(oddOccurrance([2, 3, 3, 2, 5]));
console.log(oddOccurrance([2, 3, 3, 2, 5, -2, 5]));

File: Codility/Codility-py/02-arrays/OddOccurrencesInArray.py
def solution(A):
    length = len(A)
    A.sort()

    i = 0
    while i < length:
        if i == length - 1 or A[i] != A[i + 1]:
            return A[i]
        i += 2


print(solution([9, 3, 9]))  # 3
print(solution([9, 3, 9, 3, 7, 9, 9]))  # 7
print(solution([2, 2, 2]))  # 2
print(solution([2, 2, 3]))  # 3
